xla/service/gpu/gpu_compiler_test.cc

```diff
@@ -1405,8 +1439,15 @@ TEST_F(PassOrderTest, PassesAreRunInCorrectOrder) {
                   /*last_pass_regex=*/"priority-fusion");
   VerifyPassOrder(/*first_pass_regex=*/"layout-assignment",
                   /*last_pass_regex=*/"layout_normalization");
-  VerifyPassOrder(/*first_pass_regex=*/"host-offload-legalize",
-                  /*last_pass_regex=*/"layout_normalization");
+}
+
+TEST_F(PassOrderTest, OffloadingPassesAreRunInCorrectOrder) {
+  VerifyPassRunsBefore(/*first_pass_regex=*/"host-offload-legalize",
+                       /*other_pass_regex=*/"layout_normalization");
+  auto pass_range =
+      VerifyPassOrder(/*first_pass_regex=*/"host-offload-legalize",
```

Comment:
Can you comment on what the pass range being extract here is?

Also perhaps factor out a util `GetOrderedPassesRange` with a docstring explaining what it does? Calling `VerifyPassOrder` just to extract a `PassRange` seems odd.

Reply:
This is how it is used in other places in the file.

Also, the legalizer is supposed to prepare for running the offloader, so this constraint should make sense.